src/fontinfo.rs

```diff
@@ -11,10 +13,51 @@ use crate::Error;
 // values.
 type Integer = i32;
 type NonNegativeInteger = u32;
-type IntegerOrFloat = f64;
 type Float = f64;
 type Bitlist = Vec<u8>;
 
+/// IntegerOrFloat represents a number that can be an integer or float. It should
+/// serialize to an integer if it effectively represents one.
+#[derive(Debug, Clone, PartialEq)]
+pub struct IntegerOrFloat(f64);
+
+impl IntegerOrFloat {
+    fn is_integer(&self) -> bool {
+        (self.0 - self.round()).abs() < std::f64::EPSILON
+    }
+}
```

Comment:
we're going to need some other methods here, like something for getting the actual value. Maybe just public `get` and `set` methods?